src/xnatdcat/cli_app.py

```diff
@@ -183,36 +167,99 @@ def cli_main():
         logger.debug("Verbose mode enabled")
 
-    config = load_configuration(config)
+    config = load_xnatdcat_configuration(config)
 
     if optin or optout:
         config['xnatdcat']['optin'] = optin
         config['xnatdcat']['optout'] = optout
 
     # with connect_cli(cli=False, **kwargs) as session:
-    with __connect_xnat(server, username, password) as session:
+    # If username is not environment variable and password is, that's usually not intended
+    # Thus we clear password so xnatpy can deal with it
+    if ctx.get_parameter_source("username") != click.core.ParameterSource.ENVIRONMENT:
+        if ctx.get_parameter_source("password") == click.core.ParameterSource.ENVIRONMENT:
+            password = None
+
+    ctx.obj['xnat_conn'] = __connect_xnat(server, username, password)
+    ctx.obj['config'] = config
+
+    # output_dcat(server, username, password, output, format, config)
+
+
+@cli_click.command(name='dcat')
+@click.option(
+    '-o',
+    '--output',
+    'output',
+    default=None,
+    type=click.Path(writable=True, dir_okay=False),
+    help="Destination file to write output to. If not set, the script will print serialized output to stdout.",
+)
+@click.option(
+    "-f",
+    "--format",
+    default="turtle",
+    type=click.Choice(
+        ['xml', 'n3', 'turtle', 'nt', 'pretty-xml', 'trix', 'trig', 'nquads', 'json-ld', 'hext'], case_sensitive=False
+    ),
+    help=(
+        "The format that the output should be written in. This value references a"
+        " Serializer plugin in RDFlib. Supportd values are: "
+        " \"xml\", \"n3\", \"turtle\", \"nt\", \"pretty-xml\", \"trix\", \"trig\", \"nquads\","
+        " \"json-ld\" and \"hext\". Defaults to \"turtle\"."
+    ),
+)
+@click.pass_context
+def output_dcat(ctx, output, format):
+    # , server, username, password, output, format, config):
+    config = ctx.obj['config']
+    with ctx.obj['xnat_conn'] as session:
         logger.debug('Connected to XNAT server')
         g = xnat_to_RDF(session, config)
         logger.debug('Finished acquiring RDF graph')
 
     if output:
         logger.debug("Output option set, serializing output to file %s in %s format", output, format)
         g.serialize(destination=output, format=format)
+
     else:
         logger.debug("Sending output to stdout")
         print(g.serialize(format=format))
 
-    # click.abort(0)
+
+@click.option("-f", "--fdp", envvar='XNATDCAT_FDP', type=str, required=True, help="URL of FDP to push to")
+@click.option(
+    "-u", "--username", envvar='XNATDCAT_FDP_USER', type=str, required=True, help="Username of FDP to push to"
+)
+@click.option(
+    "-p", "--password", envvar='XNATDCAT_FDP_PASS', type=str, required=True, help="Password of FDP to push to"
+)
+@click.option("-c", "--catalog", default=None, type=URIRef, help="Catalog URI of FDP")
+@cli_click.command(name='fdp')
+@click.pass_context
+def output_fdp(ctx, fdp, username, password, catalog):
```

Comment:
type hints? or does click not like that?

Reply:
Fixed